src/main/java/com/gregtechceu/gtceu/integration/jade/provider/RecipeOutputProvider.java

```diff
@@ -65,7 +69,7 @@ protected void write(CompoundTag data, RecipeLogic recipeLogic) {
                     var stack = stacks[0];
 
                     var itemTag = new CompoundTag();
-                    GTUtil.saveItemStack(stack, itemTag);
+                    GTUtil.saveIngredient((Ingredient) item.content, itemTag);
```

Comment:
you should probably something like this here (instead of serializing the entire ingredient)
```suggestion
                    if (item.content instanceof IntProviderIngredient provider) {
                        itemTag = JsonOps.INSTANCE.convertTo(NbtOps.INSTANCE, provider.toJson());
                    } else {
                        GTUtil.saveItemStack(stack, itemTag);
                    }
```